components/engine_traits/src/engines.rs

```diff
@@ -1,21 +1,43 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+use std::fmt::{self, Debug};
 
 use crate::{
-    engine::KvEngine, errors::Result, options::WriteOptions, raft_engine::RaftEngine,
+    engine::KvEngine,
+    errors::Result,
+    options::WriteOptions,
+    raft_engine::RaftEngine,
+    engine::TabletFactory,
     write_batch::WriteBatch,
 };
 
-#[derive(Clone, Debug)]
 pub struct Engines<K, R> {
     pub kv: K,
     pub raft: R,
+    tablet_kv: Option<K>,
+    pub tablets: Option<Box<dyn TabletFactory<K> + Send>>,
 }
 
 impl<K: KvEngine, R: RaftEngine> Engines<K, R> {
     pub fn new(kv_engine: K, raft_engine: R) -> Self {
         Engines {
-            kv: kv_engine,
+            kv: kv_engine.clone(),
```

Comment:
Why clone() here?

Reply:
good catch. It's not needed. The initial code needs it, but not now.